### PR TITLE
adding jem/angular-routing for routes

### PR DESCRIPTION
**Pull request by jstenhouse**
@holtbp @rantler @briandela @vicentereig 

Moved to angular-routing for routes (https://github.com/dotJEM/angular-routing). Also moved angular JS to vendor/assets.

### CODE REVIEW

File: app/assets/javascripts/templates/home.html.haml [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
 %ul
   %li
-    %a{:href => '/#/beers'} Beers
+    %a{:sref => "'beers'"} Beers
   %li
-    %a{:href => '/#/users'} Users
+    %a{:sref => "'users'"} Users
   %li
-    %a{:href => '/#/brewers'} Brewers
+    %a{:sref => "'brewers'"} Brewers
```
Comment (jstenhouse) [inline app/assets/javascripts/templates/home.html.haml:8]: This page will have JS errors until we add routes for users and brewers. Which is conveniently up next.

Comment (holtbp) [inline app/assets/javascripts/templates/home.html.haml:8]: What is with the double quotes?

Comment (jstenhouse) [inline app/assets/javascripts/templates/home.html.haml:8]: I don't know honestly. It expects it to be quoted thus. Single quotes don't work. @vicentereig do you know why it requires the quotes to be like this?

**Comment by vicentereig** [comment]
Yup, the `sref` attribute needs to be an angular expression, a string in this case. I remember having this issue also with ui-router. Copy and pasted from the docs:

```
       <ul class="nav navbar-nav">
            <li ng-class="{ active: fn.isActive('home') }">
              <a sref="'home'">Home</a>
            </li>
            <li ng-class="{ active: fn.isActive('about') }">
              <a sref="'about'">About</a>
            </li>
            <li ng-class="{ active: fn.isActive('contact') }">
              <a sref="'contact'">Contact</a>
            </li>
          </ul>
```

**Comment by vicentereig** [comment]
A little bit off topic, how do you guys populate the `$templateCache` from `assets/javascripts/templates/*.haml.html`?

**Comment by jstenhouse** [comment]
I don't think we are populating the `$templateCache` unless code in `main.js` is taking care of it for us. Or if the `angular_assets.rb` initializer does something for us.

**Pull request merged**